### DISCUSSION TITLE
How do I start and run open-webui as a service on Windows Server 2019?

### DISCUSSION THREAD

**Discussion by eugrus** [open-webui/open-webui#6126] [General]
I've successfully installed open-webui into a venv of python3.11 on Windows Server 2019

I am able to use the following .cmd script to start it:
```
call R:\openwebui\Scripts\activate
open-webui serve
```

Now I’ve created the following Windows service:
``sc.exe create openwebui binPath="R:\openwebui\startopenwebui.cmd" start=auto``

However if I do ``Start-Service openwebui``, or ``sc.exe start openwebui`` or start the newly created service from taskmgr then the conhost or taskmgr window just closes immediately. After that the service remains stopped in the taskmgr list.

Did anyone figure out a way to start and run open-webui as a service on Windows Server?

**Comment by fluxlinkage** [comment]
You can use [NSSM](https://nssm.cc/download).
Create a .bat file like this (remember to modify path):

R:
cd R:\openwebui\Scripts
CALL activate.bat
timeout /t 1
open-webui serve

Then use nssm to install a service.
Good luck!

**Comment by cyysky** [comment]
i made a automated script to install open-webui as window services 

https://medium.com/@chongyoeyat/open-webui-5-20-windows-service-installer-3f55a33a3d8d
